### PR TITLE
Fixes #12431: Create centreon plugin

### PR DESCRIPTION
**Pull request by VictorQrt**
https://www.rudder-project.org/redmine/issues/12431

Replacing previous PR: https://github.com/Normation/rudder-plugins/pull/31

### CODE REVIEW

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

File: centreon/centreon-plugin.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
```
Comment (peckpeck) [inline centreon/centreon-plugin.py]: Use python, your code must be compatible with python 2 and 3

File: centreon/centreon-plugin.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    checkIfNodeInRudderGroup(node['rudder_id'], centreon_hosts)
+
+def delHostOnRudderHook(rudderID):
+    centreon_hosts = Host()
+    if any(h['name'] == rudderID for h in centreon_hosts.list()['result']):
+        centreon_hosts.delete(rudderID)
+
+if __name__ == '__main__':
+    args = docopt(__doc__)
+    confPath = "/opt/rudder/etc/"
```
Comment (peckpeck) [inline centreon/centreon-plugin.py]: This line should be at the top of the file, and it should just contain the full path of the configuration file

**Comment by VictorQrt** [comment]
Commit modified

File: centreon/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	cp centreon-plugin.py target/bin
+
+target/etc/centreon.conf:
+	mkdir target/etc
+	cp centreon.conf target/etc
+
+target/var/rudder/tmp/rudder_nodes.json:
+	mkdir target/var
+	mkdir target/var/rudder
+	mkdir target/var/rudder/tmp
```
Comment (peckpeck) [inline centreon/Makefile]: use mkdir -p instead

File: centreon/Makefile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	cp centreon-post-acceptance.sh target/etc/hooks.d/node-post-acceptance
+
+target/etc/hooks.d/node-pre-deletion/centreon-pre-deletion.sh:
+	mkdir target/etc/hooks.d/node-pre-deletion
+	cp centreon-pre-deletion.sh target/etc/hooks.d/node-pre-deletion
+
+target/share/python/centreonapi:
+	mkdir target/share
+	mkdir target/share/python
+	cp -r centreonapi target/share/python/
```
Comment (peckpeck) [inline centreon/Makefile]: In fact you should just have the same directory structure in the repo as you want in the plugin and the make would just become a cp -r

File: centreon/centreon-plugin.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        data = requests.get(conf['RUDDER']['rudderAPIURL'] + path, headers={ 'X-API-Token': conf['RUDDER']['rudderAPIToken'] }, verify=False).json()
+    except json.decoder.JSONDecodeError:
+        print("[!] Error : check your Rudder API token")
+        sys.exit(-1)
+    return data
+
+# Used to determinate which IP will be given to Centreon, from all the IPs Rudder sends us
+def validateIPAddr(ip):
+    try:
+        socket.inet_aton(ip)
```
Comment (peckpeck) [inline centreon/centreon-plugin.py]: i don't see how this is a validation of an ip address

File: centreon/centreon-pre-deletion.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
```
Comment (peckpeck) [inline centreon/centreon-pre-deletion.sh]: I can't see from github, but all your scripts must have the executable bit before commiting then.

File: centreon/centreon.conf [hunk 7]
```diff
@@ -0,0 +1,10 @@
+[CENTREON]
+username = admin
+password = admin
+centreonWebserviceURL = http://192.168.41.2/centreon
```
Comment (peckpeck) [inline centreon/centreon.conf]: This should contain documentation comments instead of real data

**Comment by VictorQrt** [comment]
Commit modified

File: centreon/Makefile [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+# make licensed : will build a license limited version of the plugin
+#
+
+FILES = bin share/python/centreonapi var etc
+SCRIPTS = postinst
+
+include ../makefiles/common-plugin.mk
+
+target/bin:
+	mkdir target/bin
```
Comment (peckpeck) [inline centreon/Makefile]: useless mkdir

**Comment by VictorQrt** [comment]
Commit modified

File: centreon/bin/centreon-plugin.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    try:
+        data = requests.get(conf['RUDDER']['rudderAPIURL'] + path, headers={ 'X-API-Token': conf['RUDDER']['rudderAPIToken'] }, verify=False).json()
+    except json.decoder.JSONDecodeError:
+        print("[!] Error : check your Rudder API token")
+        sys.exit(-1)
+    return data
+
+# Used to determinate which IP will be given to Centreon, from all the IPs Rudder sends us
+def validateIPAddr(ip):
+    return len(ip) <= 15 and not ip == '127.0.0.1' and not ip =='0:0:0:0:0:0:0:1'
```
Comment (peckpeck) [inline centreon/bin/centreon-plugin.py]: you can do better

**Comment by VictorQrt** [comment]
Commit modified

**Pull request merged**